ci/test/00_setup_env_s390x.sh

```diff
@@ -6,9 +6,10 @@
 
 export LC_ALL=C.UTF-8
 
-export HOST=s390x-unknown-linux-gnu
-export PACKAGES="clang llvm python3-zmq qtbase5-dev qttools5-dev-tools libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev libqrencode-dev"
-export NO_DEPENDS=1
+# The host arch is unknown, so we run the tests through qemu.
+# If the host is arm and wants to run the tests natively, it can set QEMU_USER_CMD to the empty string.
+export QEMU_USER_CMD="${QEMU_USER_CMD:"qemu-s390x"}"
+export PACKAGES="python3 python3-zmq bsdmainutils qemu-user qtbase5-dev"
```

Comment:
```suggestion
export PACKAGES="python3 python3-zmq bsdmainutils qemu-user"
```

Reply:
I wrote about that in the OP

Reply:
Hmm. CC @dongcarl and @fanquake for ideas on why this is needed

Reply:
Huh... Could you try doing
```
$(package)_config_opts_s390x_linux = -platform linux-g++ -xplatform bitcoin-linux-g++
```
in `qt.mk` and seeing if that works?

Reply:
I'll try

Reply:
Haven't tested that yet, but i'm starting to think that this problem isn't deterministic :O somehow it passes now without it https://travis-ci.org/elichai/bitcoin/jobs/620711630?utm_medium=notification&utm_source=github_status

Reply:
> Haven't tested that yet, but i'm starting to think that this problem isn't deterministic :O somehow it passes now without it [travis-ci.org/elichai/bitcoin/jobs/620711630?utm_medium=notification&utm_source=github_status](https://travis-ci.org/elichai/bitcoin/jobs/620711630?utm_medium=notification&utm_source=github_status)

The commit hash in the original PR description does not match this one... So perhaps you rebased or made other changes?

Reply:
I tried cross compilation once from x86_64->s390x and it only worked with the suggestion by @dongcarl 

```diff
diff --git a/depends/packages/qt.mk b/depends/packages/qt.mk
index a9f82d038..2db7f26b2 100644
--- a/depends/packages/qt.mk
+++ b/depends/packages/qt.mk
@@ -106,6 +106,8 @@ $(package)_config_opts += -no-feature-xml
 
 $(package)_config_opts_darwin = -no-dbus
 
+$(package)_config_opts_s390x_linux = -platform linux-g++ -xplatform bitcoin-linux-g++
+
 ifneq ($(build_os),darwin)
 $(package)_config_opts_darwin += -xplatform macx-clang-linux
 $(package)_config_opts_darwin += -device-option MAC_SDK_PATH=$(OSX_SDK)
```

Full diff:

```diff
diff --git a/ci/test/00_setup_env_s390x.sh b/ci/test/00_setup_env_s390x.sh
index 8116268d3..43a820529 100644
--- a/ci/test/00_setup_env_s390x.sh
+++ b/ci/test/00_setup_env_s390x.sh
@@ -6,10 +6,11 @@
 
 export LC_ALL=C.UTF-8
 
+export HOST=s390x-linux-gnu
 # The host arch is unknown, so we run the tests through qemu.
 # If the host is s390x and wants to run the tests natively, it can set QEMU_USER_CMD to the empty string.
 export QEMU_USER_CMD="${QEMU_USER_CMD:"qemu-s390x"}"
-export PACKAGES="python3 python3-zmq bsdmainutils qemu-user qtbase5-dev"
+export PACKAGES="python3-zmq g++-s390x-linux-gnu qemu-user"
 export RUN_UNIT_TESTS=true
 export RUN_FUNCTIONAL_TESTS=true
 export GOAL="install"
diff --git a/depends/packages/qt.mk b/depends/packages/qt.mk
index a9f82d038..2db7f26b2 100644
--- a/depends/packages/qt.mk
+++ b/depends/packages/qt.mk
@@ -106,6 +106,8 @@ $(package)_config_opts += -no-feature-xml
 
 $(package)_config_opts_darwin = -no-dbus
 
+$(package)_config_opts_s390x_linux = -platform linux-g++ -xplatform bitcoin-linux-g++
+
 ifneq ($(build_os),darwin)
 $(package)_config_opts_darwin += -xplatform macx-clang-linux
 $(package)_config_opts_darwin += -device-option MAC_SDK_PATH=$(OSX_SDK)

Reply:
Note: If we are applying this diff, can we make sure we put the `$(package)_config_opts_s390x_linux` line down with the other Linux options, rather than splitting up the darwin config.

Reply:
It works without that diff. the only problem is the `HOST`.
we could do a trick similiar to `QEMU_USER_CMD` where we only apply the `HOST` var if it's in qemu